### PR TITLE
#4359 + #4360 umb editor navigation content apps anchor-dropdown

### PR DESCRIPTION
**Pull request by nielslyngsoe**
Fixing #4359 and #4360

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/common/directives/components/editor/umbeditornavigationitem.directive.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/**
+@ngdoc directive
+@name umbraco.directives.directive:umbTabContent
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/editor/umbeditornavigationitem.directive.js]: @nielslyngsoe the eagle-eyed nit-pick is that `umbTabContent` is no longer named this and the angular JS docs we generate will be wrong & mis-leading

File: src/Umbraco.Web.UI.Client/src/common/directives/components/editor/umbeditornavigationitem.directive.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+@restrict E
+@scope
+
+@description
+Use this directive to render tab content. For an example see: {@link umbraco.directives.directive:umbTabContent umbTabContent}
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/editor/umbeditornavigationitem.directive.js]: Similar as ⬆️ just make sure to update this @nielslyngsoe

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js [hunk 3]
```diff
@@ -871,13 +871,34 @@
...
+         * Call back when a content app changes
+         * @param {any} app
+         */
+        $scope.appAnchorChanged = function (app, anchor) {
+            /*
+            // This is how it should be done — but because of the current architecture, i had to give the responsibilty to 'activate' the content-app to the menu, cause thats the only place you currently can change the active content-app.
+            // proposal. Make sure that the content-app-menu dosnt do anything. Just make it callback when menu-item are clicked, then make the controller of the view handle what to be done. Then the controller should broadcast a local event, notifying that the current content-app has been changed.
+            // read more on this issue: https://github.com/umbraco/Umbraco-CMS/issues/4467
+            if ($scope.app !== app) {
+                // TODO: Not working..!!!!!!!!
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js]: Is this big block of comment @nielslyngsoe no longer needed & left in by mistake or is this still relevant?

**Comment by warrenbuckley** [comment]
@nielslyngsoe can this be re-used at all by third party developers in their own content apps?
If so do you have a simple example I can try to ensure it works & in combination if I put a colour round badge on the content app as well.

**Comment by nielslyngsoe** [comment]
> @nielslyngsoe can this be re-used at all by third party developers in their own content apps?
> If so do you have a simple example I can try to ensure it works & in combination if I put a colour round badge on the content app as well.

I have created a feature request to go through this.

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbtabbedcontent.directive.js [hunk 4]
```diff
@@ -2,40 +2,153 @@
...
+                    setActiveAnchor($args.anchor);
+                    scrollTo($args.anchor.id);
+                }
+            });
+            
+            //ensure to unregister from all dom-events
+            $scope.$on('$destroy', function () {
+                cancelScrollTween();
+                scrollableNode.removeEventListener("scroll", onScroll);
+                scrollableNode.removeEventListener("mousehweel", cancelScrollTween);
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbtabbedcontent.directive.js]:
```suggestion
                scrollableNode.removeEventListener("mousewheel", cancelScrollTween);
```

**Pull request merged**